lxd/storage_volumes.go

```diff
@@ -2729,6 +2729,110 @@ func createStoragePoolVolumeFromBackup(s *state.State, r *http.Request, requestP
 	return operations.OperationResponse(op)
 }
 
+// ctxStorageVolumeDetails is the request.CtxKey corresponding to storageVolumeDetails, which is added to the request
+// context in addStoragePoolVolumeDetailsToRequestContext.
+const ctxStorageVolumeDetails request.CtxKey = "storage-volume-details"
+
+// storageVolumeDetails contains details common to all storage volume requests. A value of this type is added to the
+// request context when addStoragePoolVolumeDetailsToRequestContext is called. We do this to avoid repeated logic when
+// parsing the request details and/or making database calls to get the storage pool or effective project. These fields
+// are required for the storage volume access check, and are subsequently available in the storage volume handlers.
+type storageVolumeDetails struct {
+	volumeName           string
+	volumeTypeName       string
+	volumeType           int
+	pool                 storagePools.Pool
+	requestProjectName   string
```

Comment:
This field seems like something we should have a common context field for, rather than embedding into every entity type specific context structure?

Also the `effectiveProjectName` could also be one that isn't specific to the entity type, but rather will be set by the middleware handler based on the type of entity in the request and the project settings.